src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -197,16 +201,39 @@ private async Task LoadAsync()
             {
                 foreach (var secretDescriptor in secretDescriptors)
                 {
-                    var result = await client.GetSecretAsync(store, secretDescriptor.SecretName, secretDescriptor.Metadata).ConfigureAwait(false);
+                    
+                    Dictionary<string, string> result;
+
+                    try
+                    {
+                        result = await client
+                            .GetSecretAsync(store, secretDescriptor.SecretKey, secretDescriptor.Metadata)
+                            .ConfigureAwait(false);
+                    }
+                    catch (DaprException e)
+                    {
+                        if (secretDescriptor.IsRequired)
+                        {
+                            throw e;
+                        }
+                            
+                        Console.WriteLine($"'{secretDescriptor.SecretName}' doesn't exists in Key Vault but because " +
+                                          $"it doesn't require existence so all errors suppressed!");
+
+                        result = new Dictionary<string, string>();
+                    }
 
                     foreach (var key in result.Keys)
                     {
                         if (data.ContainsKey(key))
                         {
-                            throw new InvalidOperationException($"A duplicate key '{key}' was found in the secret store '{store}'. Please remove any duplicates from your secret store.");
+                            throw new InvalidOperationException($"A duplicate key '{key}' was found in the " +
+                                                                $"secret store '{store}'. Please remove any " +
+                                                                $"duplicates from your secret store.");
                         }
 
-                        data.Add(normalizeKey ? NormalizeKey(key) : key, result[key]);
+                        data.Add(normalizeKey ? NormalizeKey(secretDescriptor.SecretName) : secretDescriptor.SecretName, 
+                            result[key]);
```

Comment:
You're doing a lot of line splitting here. What do you have your margin set to? 120-160 characters is a good width for modern screens (I prefer 160).

Reply:
Fixed